src/sentry/api/endpoints/data_export.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import absolute_import
+
+from sentry import features
+from sentry.api.bases.incident import IncidentPermission
+from sentry.api.base import Endpoint
+from django.utils import timezone
+
+
+from sentry.tasks.data_export import create_record
+
+from sentry.models import ExportedData, Organization
+
+
+class DataExportEndpoint(Endpoint):
+    permission_classes = (IncidentPermission,)
+
+    def get(self, *args, **kwargs):
+        """
+        Retrieve information about the temporary file record.
+        Used to populate page emailed to the user.
+        """
+        organization = Organization.objects.get(slug=kwargs["organization_slug"])
+        if not features.has("organizations:data-export", organization):
+            return self.respond(status=404)
+        try:
+            export_record = ExportedData.objects.get(data_id=kwargs["data_id"])
+            if export_record.finished_at is None:
+                return self.respond({"status": "EARLY"})
+            elif export_record.expired_at < timezone.now():
+                return self.respond({"status": "EXPIRED"})
```

Comment:
It might also make sense to have a `status` property on the `ExportData` model, which returns a value from an enum like `ExportStatus` which would have values `EARLY`, `READY`, `EXPIRED` (or something along those lines). Then you don't need this logic here, you can just check the statuses instead.

Reply:
Since this has to do with what time the request is made, it can't be saved in the model itself (I.e. accessing the same object at two different times = two diff results)

Reply:
You can have this logic as a `@property` method on the model and it'll calculate it.

Reply:
Learning new things everyday lol, thanks i'll do that!